docs/application-developers-reference-guide.md

```diff
@@ -0,0 +1,83 @@
+
+## Introduction
+
+BKPR is a curated collection of services running on top of your existing Kubernetes cluster with the aim of automating the configuration of public access, logging and monitoring, and management of DNS records and TLS certificates.
+
+For BKPR to perform this automation, you need to apply a few configurations in Kubernetes manifests of your application. The following sections walk you through each of these configurations.
+
+## Ingress and Let's Encrypt
+
+BKPR installs an [NGINX controller](https://github.com/kubernetes/ingress-nginx) which provides load balancing, SSL termination and name-based virtual hosting for your applications. Ingress enables application developers to expose HTTP and HTTPS routes from outside the cluster to services running inside the cluster.
+
+BKPR performs DNS zone record updates for host entries present in [ingress rules](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-rules) automatically without any additional configurations.
+
+To automate the Let's Encrypt certificate provisioning you should add the annotation `kubernetes.io/tls-acme: "true"` to the ingress resource of your application. Optionally you could add the annotation `kubernetes.io/ingress.class: "nginx"` to designate that the ingress resource should only be handled by the NGINX controller.
+
+For example, with the following snippet for a k8s ingress resource, BKPR will automatically update the DNS records for `myapp.mydomain.com` and request Let's Encrypt for a valid TLS certificate for your application, following which you would be able to access the application securely over the Internet.
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: "myapp-ingress"
+  labels:
+    name: "myapp"
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    kubernetes.io/tls-acme: "true"
+spec:
+  tls:
+  - hosts:
+    - myapp.mydomain.com
+    secretName: myapp-tls
+  rules:
+  - host: myapp.mydomain.com
+    http:
+      paths:
+        - path: /
+          backend:
+            serviceName: myapp-svc
+            servicePort: 80
+```
+
+## Logging
+
+BKPR installs [Elasticsearch](https://elastic.co/products/elasticsearch) in the cluster which provides a RESTful search and analytics engine for processing your application logs. The stack uses [Fluentd](https://www.fluentd.org/) to capture the standard output of every container running in the cluster and persistent storage to store historical data of the cluster container logs.
+
+For Fluentd to be able to capture your applications logs, as a cloud-native application developer, all you need to do is configure your application to output the logs to the standard output of the application container.
```

Comment:
Perhaps it would be nice to mention the types of log records that our Fluentd supports out-of-the box (e.g. Apache, etc.).

Reply:
do you know if we have a list somewhere?

Reply:
But honestly in this doc we do not intend go into details, we only intend to provide a quick reference what needs to be done by the app author to get the mundane stuff automated by BKPR

Reply:
I think this is documented in Fluentd's documentation. Mine was a suggestion. Feel free to ignore it if you think it is too much detail, but I think it's pretty cool to say that if running Apache as a container under BKPR means your Apache logs get properly parsed and fields detected automatically :-)